Use pivot.service to perform pivot

jlebon opened the pull request:
Rather than directly executing `pivot` from within the container, use
its new service unit to run it. The main advantage of this is that
`pivot` can correctly run `ostree` as `install_t` which is required for
writing new SELinux labels. More generally though, we should be running
such host tools within the host context rather than from the container.

This works nicely though the downside is that we lose progress output
from pivot. I'd like to add journal proxying to fix that.

Closes: #314 

Requires: https://github.com/openshift/pivot/pull/25

ashcrow commented:
Infra flakes

/retest

ashcrow commented:
Infra flake

/retest

jlebon commented:
OK, this now also proxies journal logs from the pivot unit:

```
I0130 22:19:17.642994   44729 update.go:643] Updating OS to registry.svc.ci.openshift.org/rhcos/maipo@sha256:ede3888e50016d61a720af2fe3f80e67e86bd819e16516ac36538456d46e0d77
pivot.service: I0130 22:19:20.009133   44951 root.go:89] Resolved to: registry.svc.ci.openshift.org/rhcos/maipo@sha256:ede3888e50016d61a720af2fe3f80e67e86bd819e16516ac36538456d46e0d77
pivot.service: I0130 22:19:20.009160   44951 root.go:103] Pivoting to: 47.198 (66e5fd101880c1583362e6ec363fd24c630290ab68a126b3e614143a1c1a4cda)
pivot.service: I0130 22:19:20.009173   44951 run.go:15] Running: podman pull registry.svc.ci.openshift.org/rhcos/maipo@sha256:ede3888e50016d61a720af2fe3f80e67e86bd819e16516ac36538456d46e0d77
```

(Notice the `pivot.service:` prefix in the above, those are from the journal).

Though one thing I'm unsure of is:

> One note here is that this requires `CGO_ENABLED=1` because the journal
bindings use the dlopen package, which has `-ldl`.

I'm not sure of the ramifications of this. IIUC, it's safe to enable this since we're not cross-compiling, right?

jlebon commented:
```
vendor/github.com/coreos/go-systemd/sdjournal/journal.go:2...atal error: systemd/sd-journal.h: No such file or directory
 // #include <systemd/sd-journal.h>
                                 ^
compilation terminated.
error: build error: running 'WHAT=machine-config-controller ./hack/build-go.sh' failed with exit code 2
```

Ahh right, this will require adding `systemd-devel` as a buildreq.

ashcrow commented on hack/build-go.sh:
```diff
@@ -31,4 +31,4 @@ fi
 
 echo "Building ${REPO}/cmd/${WHAT} (${VERSION_OVERRIDE})"
-CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH} go build ${GOFLAGS} -ldflags "${GLDFLAGS}" -o ${BIN_PATH}/${WHAT} ${REPO}/cmd/${WHAT}
+CGO_ENABLED=1 GOOS=${GOOS} GOARCH=${GOARCH} go build ${GOFLAGS} -ldflags "${GLDFLAGS}" -o ${BIN_PATH}/${WHAT} ${REPO}/cmd/${WHAT}
```
CGO :frowning: ... but understandable.

ashcrow commented on pkg/daemon/rpm-ostree.go:
```diff
@@ -6,10 +6,14 @@ import (
+	"github.com/golang/glog"
 )
 
+const pivotUnit = "pivot.service"
```
nit:
```golang
const (
    pivotUnit = "pivot.service"
    etcPivotFile = "/etc/pivot/image-pullspec"
)
```

jlebon commented:
> Building fails due to lack of C headers:

Yup, mentioned this higher up in https://github.com/openshift/machine-config-operator/pull/335#issuecomment-459137816. :) Just waiting to see if someone more versed in golang than I can confirm `CGO_ENABLED=1` is indeed fine before fixing things up.

jlebon commented:
@abhinavdahiya Any reason why [this commit](https://github.com/openshift/machine-config-operator/commit/3792c944f2ced1be4fe8c2758f390d097a0b7927) added `CGO_ENABLED=0`? Were we vendoring something which linked to something we didn't want?

abhinavdahiya commented:
> @abhinavdahiya Any reason why [this commit](https://github.com/openshift/machine-config-operator/commit/3792c944f2ced1be4fe8c2758f390d097a0b7927) added `CGO_ENABLED=0`? Were we vendoring something which linked to something we didn't want?

To force static binaries. Without it @crawford  was unable to test binaries locally.


Also don't turn on cgo for all components, I think we should do it only for the daemon.

jlebon commented on pkg/daemon/rpm-ostree.go:
```diff
@@ -84,5 +94,84 @@ func (r *RpmOstreeClient) GetBootedOSImageURL(rootMount string) (string, string,
 // osImageURL. See https://github.com/openshift/pivot.
 func (r *RpmOstreeClient) RunPivot(osImageURL string) error {
-	return Run("/bin/pivot", osImageURL)
+	if err := ioutil.WriteFile(etcPivotFile, []byte(osImageURL), 0644); err != nil {
```
Yup, thanks for helping test this!

crawford commented:
> Without it @crawford was unable to test binaries locally.

I run NixOS, which is a really good way of double checking if you've made too many assumptions ;) I don't have libc, bash, or even a dynamic linker in the traditional location. I'm on a personal crusade to kill the misuse of the term "Linux Binary" (looking right at you, https://golang.org/dl/).

jlebon commented:
OK, comments addressed! This is mostly waiting for https://github.com/openshift/release/pull/2783 now.

ashcrow commented:
Added more to the pre-req for review ... also ping'd owners in chat. :crossed_fingers:

ashcrow commented:
https://github.com/openshift/release/pull/2783 merged

/retest

runcom commented:
golang base image hasn't probably propagated yet (or built fwiw)

runcom commented:
Clayton said the base golang image should be rebuild now, trying...
@jlebon does this still need WIP? 

/retest

jlebon commented:
Yeah, want to retest it one last time now that all the prereqs are baked into the latest RHCOS image. But need to set up a new cluster first.

ashcrow commented:
/retest

runcom commented:
looks like image has been picked up by tests now, retesting rhel-images now

/test rhel-images

ashcrow commented:
```
vendor/github.com/coreos/go-systemd/sdjournal/journal.go:2...atal error: systemd/sd-journal.h: No such file or directory
 // #include <systemd/sd-journal.h>
```

jlebon commented:
OK, dropping WIP on this! This is ready for review, but we need to hold it until the latest RHCOS has an updated `pivot` and `redhat-release-coreos`.

/hold

jlebon commented:
So with #396 fixed, this should compile now.
/retest

runcom commented:
/hold cancel

jlebon commented:
(No functional changes, I just updated the commit messages to reflect reality.)

jlebon commented:
> /hold cancel

To be clear, the /hold I did earlier wasn't because of #396, it was because this new code requires a newer RHCOS we're still waiting on (it used to be new enough, but then some components rolled back to older versions as we rewired some things in the back). So just gonna re-add it for now :)

/hold

runcom commented:
Alrighty, sorry for the hold cancel then, left a couple of comments but looks great otherwise

ashcrow commented:
/retest

runcom commented:
/retest

ashcrow commented:
/retest

jlebon commented:
Comments addressed!

And latest RHCOS now contains everything we need, so dropping hold.
/hold cancel

Sample output:

```
I0209 15:35:31.404039   89314 update.go:643] Updating OS to registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825
pivot.service: I0209 15:35:34.044344   90592 root.go:89] Resolved to: registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825
pivot.service: I0209 15:35:34.044382   90592 run.go:16] Running: podman pull registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825
pivot.service: Trying to pull registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825...Getting image source signatures
Copying blob 8c7755974db3: 15.97 MiB / 591.21 MiBB1.21 MiB
I0209 15:35:40.929229   89314 daemon.go:338] Kubelet health running
Copying blob 8c7755974db3: 169.03 MiB / 591.21 MiBet health ok
I0209 15:36:10.940125   89314 daemon.go:338] Kubelet health running
Copying blob 8c7755974db3: 293.25 MiB / 591.21 MiBet health ok
Copying blob 8c7755974db3: 320.05 MiB / 591.21 MiB in 60 seconds
I0209 15:36:40.958742   89314 daemon.go:338] Kubelet health running
Copying blob 8c7755974db3: 472.97 MiB / 591.21 MiBet health ok
I0209 15:37:10.966013   89314 daemon.go:338] Kubelet health running
Copying blob 8c7755974db3: 591.21 MiB / 591.21 MiB  1m57sth ok
Copying config 0bc029ca8655: 412 B / 412 B  0sB / 412 B
pivot.service: Writing manifest to image destination
pivot.service: Storing signatures
I0209 15:37:40.970021   89314 daemon.go:338] Kubelet health running
I0209 15:37:40.984082   89314 daemon.go:365] Kubelet health ok
pivot.service: 0bc029ca86559a65f60d96ec63b4c6270d7bf97d8833641775e2d85abb057e3a
pivot.service: I0209 15:37:52.193805   90592 run.go:16] Running: podman kill ostree-container-pivot
pivot.service: unable to find container ostree-container-pivot: no container with name or ID ostree-container-pivot found: no such container
pivot.service: W0209 15:37:52.321256   90592 run.go:69] (ignored) podman: exit status 125
pivot.service: I0209 15:37:52.321288   90592 run.go:16] Running: podman rm -f ostree-container-pivot
pivot.service: unable to find container ostree-container-pivot: no container with name or ID ostree-container-pivot found: no such container
pivot.service: W0209 15:37:52.552891   90592 run.go:69] (ignored) podman: exit status 125
pivot.service: I0209 15:37:52.552932   90592 run.go:16] Running: podman create --net=none --name ostree-container-pivot registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825
pivot.service: I0209 15:37:53.010863   90592 run.go:16] Running: podman mount e123eb12c608ea99657c14f25c11c035c360f78c3a240f8d082584962e27452a
pivot.service: I0209 15:37:53.308285   90592 root.go:114] Pivoting to: 47.310 (e223c0a6573b750c8b890d1f4e6ddc32ffaf90124c1a96b05b03ca0cc27020f6)
pivot.service: I0209 15:37:53.308340   90592 run.go:16] Running: rpm-ostree rebase --experimental /var/lib/containers/storage/overlay/5a09e87ffa9cafc18db4cf51e2a61a7ffc0aa3b9385a8156c3617292cd039421/merged/srv/repo:e223c0a6573b750c8b890d1f4e6ddc32ffaf90124c1a96b05b03ca0cc27020f6 --custom-origin-url pivot://registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825 --custom-origin-description Managed by pivot tool
rpm-ostreed.service: Reading config file '/etc/rpm-ostreed.conf'
rpm-ostreed.service: In idle state; will auto-exit in 62 seconds
rpm-ostreed.service: client(id:cli dbus:1.750 unit:pivot.service uid:0) added; new total=1
rpm-ostreed.service: Initiated txn UpdateDeployment for client(id:cli dbus:1.750 unit:pivot.service uid:0): /org/projectatomic/rpmostree1/redhat_coreos_maipo
pivot.service: Writing objects: 6...done
pivot.service: Staging deployment...done
rpm-ostreed.service: sanitycheck(/usr/bin/true) successful
rpm-ostreed.service: Txn UpdateDeployment on /org/projectatomic/rpmostree1/redhat_coreos_maipo successful
pivot.service: Freed: 186.4 MB (pkgcache branches: 0)
pivot.service: Upgraded:
pivot.service: atomic-openshift-clients 4.0.0-0.163.0.git.0.2abef44.el7 -> 4.0.0-0.167.0.git.0.2d4963f.el7
pivot.service: atomic-openshift-hyperkube 4.0.0-0.163.0.git.0.2abef44.el7 -> 4.0.0-0.167.0.git.0.2d4963f.el7
pivot.service: atomic-openshift-node 4.0.0-0.163.0.git.0.2abef44.el7 -> 4.0.0-0.167.0.git.0.2d4963f.el7
pivot.service: pivot 0.0.2.18-fb4c73f5cd9f552059b5e35377b2760224416b0f -> 0.0.3-1.el7
pivot.service: redhat-release-coreos 4.0-20180515.0.atomic.el7.0 -> 4.0-20190207.0.atomic.el7.0
pivot.service: runc 1.0.0-58.dev.rhaos4.0.git2abd837.el7 -> 1.0.0-60.dev.rhaos4.0.git2abd837.el7
pivot.service: Downgraded:
pivot.service: nss-altfiles 2.18.1-11.el7 -> 0-2.atomic.git20131217gite2a80593.el7
pivot.service: ostree 2019.1.5-6649032a375238255052a43adb8bc56faac989ca.bcbd1d97af9663b14119357388267e2c217c7112.el7 -> 2019.1-1.el7_6
pivot.service: ostree-grub2 2019.1.5-6649032a375238255052a43adb8bc56faac989ca.bcbd1d97af9663b14119357388267e2c217c7112.el7 -> 2019.1-1.el7_6
pivot.service: rpm-ostree 2019.1.8-9ca74d0c31eb69e8022aa6846a57ed11de817291.fa4bb14da2bf67181894fa561d54f6ccce91c95d.el7 -> 2019.1-3.atomic.el7
pivot.service: rpm-ostree-libs 2019.1.8-9ca74d0c31eb69e8022aa6846a57ed11de817291.fa4bb14da2bf67181894fa561d54f6ccce91c95d.el7 -> 2019.1-3.atomic.el7
pivot.service: Removed:
pivot.service: bubblewrap-0.3.1.6-94147e233fe200d1fe43a9a18c52475188b22798.el7.centos.x86_64
pivot.service: ostree-libs-2019.1.5-6649032a375238255052a43adb8bc56faac989ca.bcbd1d97af9663b14119357388267e2c217c7112.el7.x86_64
pivot.service: Added:
pivot.service: ostree-fuse-2019.1-1.el7_6.x86_64
pivot.service: Run "systemctl reboot" to start a reboot
rpm-ostreed.service: client(id:cli dbus:1.750 unit:pivot.service uid:0) vanished; remaining=0
pivot.service: I0209 15:38:04.224262   90592 run.go:16] Running: podman kill ostree-container-pivot
rpm-ostreed.service: In idle state; will auto-exit in 64 seconds
pivot.service: can only kill running containers: container state improper
pivot.service: W0209 15:38:04.374251   90592 run.go:69] (ignored) podman: exit status 125
pivot.service: I0209 15:38:04.374276   90592 run.go:16] Running: podman rm -f ostree-container-pivot
pivot.service: e123eb12c608ea99657c14f25c11c035c360f78c3a240f8d082584962e27452a
pivot.service: I0209 15:38:04.726515   90592 run.go:16] Running: podman rmi registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825
pivot.service: 0bc029ca86559a65f60d96ec63b4c6270d7bf97d8833641775e2d85abb057e3a
I0209 15:38:06.558582   89314 update.go:75] Update prepared; draining the node
```

jlebon commented on pkg/daemon/rpm-ostree.go:
```diff
@@ -84,5 +99,105 @@ func (r *RpmOstreeClient) GetBootedOSImageURL(rootMount string) (string, string,
+			return fmt.Errorf("Error queuing start job; got %s", st)
+		}
+	case <-time.After(5 * time.Minute):
+		return fmt.Errorf("Timed out waiting for start job")
```
Fixed!

runcom commented:
not sure why `rhel-images` fail in the CI for jouranld, I thought we added systemd-devel to the base `registry.svc.ci.openshift.org/openshift/release:golang-1.10` already and indeed I can see the C header file. It also fails locally for me when doing `make deploy-daemon` with the very same error (I'm missing something stupid probably)

runcom commented:
trying an upgrade to the latest oscontainer for ootpa, I get this:
```
20:08:05 [github.com/openshift/installer] ‹master*› oc logs -p machine-config-daemon-qghbm
I0209 19:01:35.548851   33639 start.go:52] Version: 3.11.0-586-g3f77d419
I0209 19:01:35.549179   33639 start.go:88] starting node writer
I0209 19:01:35.557321   33639 run.go:22] Running captured: chroot /rootfs rpm-ostree status --json
I0209 19:01:35.645087   33639 daemon.go:155] Booted osImageURL: registry.svc.ci.openshift.org/rhcos/maipo@sha256:def5c8ef775f634d99e3f603da0db7591a0113ad8307bc8d8545a170787a2825 (47.310)
I0209 19:01:35.645278   33639 daemon.go:227] Managing node: ip-10-0-175-8.us-west-2.compute.internal
I0209 19:01:35.658271   33639 start.go:146] Calling chroot("/rootfs")
I0209 19:01:35.671963   33639 daemon.go:432] Current+desired config: worker-bcb39e51c5bfba89da68ec52036f119e
I0209 19:01:35.671995   33639 daemon.go:786] No target osImageURL provided
I0209 19:01:35.672722   33639 daemon.go:547] Validated on-disk state
I0209 19:01:35.672740   33639 daemon.go:579] In desired config worker-bcb39e51c5bfba89da68ec52036f119e
I0209 19:01:35.672766   33639 start.go:165] Starting MachineConfigDaemon
I0209 19:01:35.672776   33639 daemon.go:248] Enabling Kubelet Healthz Monitor
I0209 19:06:06.374178   33639 update.go:110] Checking reconcilable for config worker-bcb39e51c5bfba89da68ec52036f119e to worker-de933fa1626a5b1943a242550a96e387
I0209 19:06:06.374211   33639 update.go:149] Checking if configs are reconcilable
I0209 19:06:06.374240   33639 update.go:297] Updating files
I0209 19:06:06.374250   33639 update.go:507] Writing file "/etc/containers/registries.conf"
I0209 19:06:06.376011   33639 update.go:507] Writing file "/etc/sysconfig/crio-network"
I0209 19:06:06.377286   33639 update.go:507] Writing file "/var/lib/kubelet/config.json"
I0209 19:06:06.378569   33639 update.go:507] Writing file "/etc/kubernetes/ca.crt"
I0209 19:06:06.379822   33639 update.go:507] Writing file "/etc/sysctl.d/forward.conf"
I0209 19:06:06.381061   33639 update.go:507] Writing file "/etc/kubernetes/kubelet-plugins/volume/exec/.dummy"
I0209 19:06:06.381683   33639 update.go:507] Writing file "/etc/kubernetes/kubelet.conf"
I0209 19:06:06.382953   33639 update.go:441] Writing systemd unit "kubelet.service"
I0209 19:06:06.383092   33639 update.go:479] Enabling systemd unit "kubelet.service"
I0209 19:06:06.383271   33639 update.go:388] /etc/systemd/system/multi-user.target.wants/kubelet.service already exists. Not making a new symlink
I0209 19:06:06.383292   33639 update.go:316] Deleting stale data
I0209 19:06:06.383307   33639 update.go:603] Writing SSHKeys at "/home/core/.ssh"
I0209 19:06:06.384459   33639 update.go:643] Updating OS to registry.svc.ci.openshift.org/rhcos/ootpa@sha256:66e45253e6e7897a409f1a5faed208f215f44331dafffd2293f2e3b61f1f3435
pivot.service: I0209 19:06:10.799839   37987 root.go:89] Resolved to: registry.svc.ci.openshift.org/rhcos/ootpa@sha256:66e45253e6e7897a409f1a5faed208f215f44331dafffd2293f2e3b61f1f3435
pivot.service: I0209 19:06:10.799874   37987 run.go:16] Running: podman pull registry.svc.ci.openshift.org/rhcos/ootpa@sha256:66e45253e6e7897a409f1a5faed208f215f44331dafffd2293f2e3b61f1f3435
pivot.service: Trying to pull registry.svc.ci.openshift.org/rhcos/ootpa@sha256:66e45253e6e7897a409f1a5faed208f215f44331dafffd2293f2e3b61f1f3435...Getting image source signatures
Copying blob 2a89ffbc2816: 543.98 MiB / 543.98 MiB  21sMiB
Copying config b468bc463acf: 412 B / 412 B  0sB / 412 B 
pivot.service: Writing manifest to image destination
pivot.service: Storing signatures
pivot.service: b468bc463acfcce89afec62b34535352c7ae7b513db7ad79ad97df8844e06266
pivot.service: I0209 19:06:51.200223   37987 run.go:16] Running: podman kill ostree-container-pivot
pivot.service: unable to find container ostree-container-pivot: no container with name or ID ostree-container-pivot found: no such container
pivot.service: W0209 19:06:51.242429   37987 run.go:69] (ignored) podman: exit status 125
pivot.service: I0209 19:06:51.242482   37987 run.go:16] Running: podman rm -f ostree-container-pivot
pivot.service: unable to find container ostree-container-pivot: no container with name or ID ostree-container-pivot found: no such container
pivot.service: W0209 19:06:51.338267   37987 run.go:69] (ignored) podman: exit status 125
pivot.service: I0209 19:06:51.338320   37987 run.go:16] Running: podman create --net=none --name ostree-container-pivot registry.svc.ci.openshift.org/rhcos/ootpa@sha256:66e45253e6e7897a409f1a5faed208f215f44331dafffd2293f2e3b61f1f3435
pivot.service: I0209 19:06:51.471997   37987 run.go:16] Running: podman mount 9eb7c6b28a7c5b03cf32dd7605d6e2516d5f1db05f4de7030f1991c413638656
pivot.service: I0209 19:06:51.591874   37987 root.go:114] Pivoting to: 48.254 (862a286d84c486f9accae3af5354389e4f9bfebdfb559392b537ff4cddbd244f)
pivot.service: I0209 19:06:51.591908   37987 run.go:16] Running: rpm-ostree rebase --experimental /var/lib/containers/storage/overlay/d624160f79436adc1754c78cdf6b971cd022f8113a36f69ba350f10fcd7330e2/merged/srv/repo:862a286d84c486f9accae3af5354389e4f9bfebdfb559392b537ff4cddbd244f --custom-origin-url pivot://registry.svc.ci.openshift.org/rhcos/ootpa@sha256:66e45253e6e7897a409f1a5faed208f215f44331dafffd2293f2e3b61f1f3435 --custom-origin-description Managed by pivot tool
rpm-ostreed.service: client(id:cli dbus:1.184 unit:pivot.service uid:0) added; new total=1
rpm-ostreed.service: client(id:cli dbus:1.184 unit:pivot.service uid:0) vanished; remaining=0
rpm-ostreed.service: In idle state; will auto-exit in 62 seconds
pivot.service: error: The connection is closed
pivot.service: F0209 19:06:51.668761   37987 run.go:62] rpm-ostree: exit status 1
I0209 19:06:52.456120   33639 daemon.go:649] Unable to apply update: Failed to run pivot: pivot service did not exit successfully
E0209 19:06:52.456573   33639 writer.go:85] Marking degraded due to: Failed to run pivot: pivot service did not exit successfully
F0209 19:06:52.466403   33639 start.go:170] failed to run: Failed to run pivot: pivot service did not exit successfully
```

jlebon commented:
Hmm, can you check the journal for errors, e.g. `journalctl -u rpm-ostreed`?

runcom commented:
ok, found this after scraping the journal at around the time pivot failed:
```
Feb 10 21:44:37 ip-10-0-159-35 kernel: type=1400 audit(1549835077.480:8): avc:  denied  { read } for  pid=4040 comm="dbus-daemon" path="/var/lib/containers/storage/overlay/a3814813e35e2c58b629fb864e957a7856ac50c
```

runcom commented:
alright, that seems to be the case why it still fails on ootpa, if I `setenforce 0`, upgrade to ootpa goes through

runcom commented:
we can follow up on ootpa btw, maipo upgrade went well for me

/approve

jlebon commented:
> alright, that seems to be the case why it still fails on ootpa, if I `setenforce 0`, upgrade to ootpa goes through

Ahh yup, that's https://bugzilla.redhat.com/show_bug.cgi?id=1672404 (see also https://github.com/projectatomic/rpm-ostree/pull/1732#issuecomment-460419495). The dev rpm-ostree [uses `--disable-dfd-over-dbus`](https://src.fedoraproject.org/rpms/rpm-ostree/pull-request/28), but I had forgotten to do the same internally. Just submitted a new build now so hopefully it's in tomorrow's spin.

jlebon commented:
Hmm, looks like `ci/prow/rhel-images` is still failing due to the sdjournal dependency. Will have to look into that tomorrow.

runcom commented:
Yeah, that image still doesn't have systemd-devel and the base image is different, even if I thought it's just mirrored.
```
FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.10 AS builder
```

ashcrow commented:
> Yeah, that image (the rhel one in this repo for the daemon) still doesn't have systemd-devel and the base image is different, even if I thought it's just mirrored.
> 
> ```
> FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.10 AS builder
> ```

Is there an open issue to fix that ... or should it flow into it and we just need to bug someone to speed up the change?

runcom commented:
> Is there an open issue to fix that ... or should it flow into it and we just need to bug someone to speed up the change?

I was going to bug someone already actually as my understanding was that rhel builder images are just mirrored

runcom commented:
/retest

ashcrow commented:
```
vendor/github.com/coreos/go-systemd/sdjournal/journal.go:2...atal error: systemd/sd-journal.h: No such file or directory
```
/cc @sosiouxme

ashcrow commented:
/test rhel-images

I don't think the image is updated yet but let's try just in case.

cgwalters commented:
We could fall back to just running `journalctl` as a subprocess.

ashcrow commented:
@cgwalters true. If this test fails again it's worth the change unless @sosiouxme believes the image will be fixed today.

runcom commented:
looks like it's built now?

/retest

cgwalters commented:
/lgtm

jlebon commented:
>  :heavy_check_mark:  **ci/prow/rhel-images** — Job succeeded.

:confetti_ball:
